actionview/lib/action_view/template/types.rb

```diff
@@ -28,30 +28,30 @@ def initialize(symbol)
           @symbol = symbol.to_sym
         end
 
-        delegate :to_s, :to_sym, :to => :symbol
+        def to_s
+          @symbol.to_s
+        end
         alias to_str to_s
 
         def ref
-          to_sym || to_s
+          @symbol
         end
+        alias to_sym ref
 
         def ==(type)
-          return false if type.blank?
-          symbol.to_sym == type.to_sym
+          @symbol == type.to_sym unless type.blank?
         end
       end
 
-      cattr_accessor :type_klass
-
-      def self.delegate_to(klass)
-        self.type_klass = klass
-      end
-
-      delegate_to Type
+      cattr_accessor(:type_klass) { Type }
 
       def self.[](type)
         type_klass[type]
       end
+
+      def symbols
+        type_klass::SET.symbols
```

Comment:
Yep, that was just an example. We'll still need to convert the Set to an array though.